### ISSUE TITLE
 Reuse the TCP model in the BMP mode: Groupings

### ISSUE THREAD

**Issue by michael-scharf** [mjethanandani/ietf-tcp#86]
Comment 2 from Camilo Cardona:

However, we would like to integrate the “common”, and “statistics” sections of ietf-tcpm, but, they are containers, so we really do not know what to do besides recreating it.  

And a related feedback from Tom Petch:

and to complicate it for most other users.  A YANG grouping, like a 
procedure or subroutine, can make sense when it is well-defined, its 
meaning or function self-evident.  Most grouping in IETF models do not 
qualify and mean the user has to jump backwards and forwards, perhaps in 
and out of multiple documents, while losing the thread of what it is 
that is being modelled.

YANG has augment, a much superior way of building on the work of others 
so adding additional node as augment points is useful, carving a model 
up into disjoint pieces less so.

**Comment by michael-scharf** [comment]
@mjethanandani : What is your take on this. Could/should we use groupings?

Candidates for groupings would be:
- Definition of a TCP connection
- Statistics

**Comment by mjethanandani** [comment]
@michael-scharf, I am not a big fan of groupings, specially when they are used to duplicate the code without understanding the why.

I was queried by the authors of https://datatracker.ietf.org/doc/html/draft-cptb-grow-bmp-yang, but I was not sure what they were asking for. Now it is apparent.

My issue is not with creating groupings, it is how they will be used. The BMP protocol is monitoring BGP sessions, which are running over TCP connections. An implementation of the TCP YANG model would give them the same information which they would have (unnecessarily) duplicated in their model, without getting the context and the benefit of TCP session information. Why do they need to duplicate these statistics or connection information in their model?

**Comment by michael-scharf** [comment]
Well, I cannot speak for the authors. They have not detailed why they really would need groupings.

After thinking a bit further, I think that the current statistics in our model are aggregated for the whole TCP stack and at least for some counters it is not apparent how they would be used in other context, such as per-connection or per-application. For instance, it does not make any sense to count TCP RSTs ("leaf out-resets") for a single connection. For an application, one would probably count application-level failures instead. As a result, I agree that it is not exactly clear why statistics would need to be identically duplicated elsewhere.

A slightly more complex question is whether the modeling of a TCP connection could be reused elsewhere. I guess that some basic 5-tuple information such as the two IP addresses and the two port numbers will be used in many YANG models. Some consistency e.g. regarding the leaf names may be useful across models. So, the 5-tuple-information could be a candidate for a grouping in a basic TCP YANG model. But beyond that it is much less obvious whether other models would indeed exactly use the model defined in this document. Even the TCP connection state is probably not very useful outside the context of this model. So possibly a grouping would be very basic and just include IP addresses and port numbers, Such a grouping could be done, but it is also not particularly problematic if other YANG models define this themselves.

In a nutshell, the actual benefit of changing the model seems a bit vague, and I'll not champion a change.

Maybe one way to sort this out is the keep the model as-is for -07. We can in addition ask the authors of  https://datatracker.ietf.org/doc/html/draft-cptb-grow-bmp-yang to better reason their use case.

**Comment by mjethanandani** [comment]
Ok. Will keep this issue open, while going ahead with publication of -07 version of the model.

**Comment by michael-scharf** [comment]
I close this issue due to lack of further input from the authors of the proposed BMP model
